include/aws/common/array_list.inl

```diff
@@ -98,23 +101,30 @@ void aws_array_list_clean_up(struct aws_array_list *AWS_RESTRICT list) {
     list->length = 0;
     list->data = NULL;
     list->alloc = NULL;
+    AWS_POSTCONDITION(aws_array_list_is_valid(list));
 }
 
 AWS_STATIC_IMPL
 int aws_array_list_push_back(struct aws_array_list *AWS_RESTRICT list, const void *val) {
+    AWS_PRECONDITION(aws_array_list_is_valid(list));
     int err_code = aws_array_list_set_at(list, val, aws_array_list_length(list));
 
     if (err_code && aws_last_error() == AWS_ERROR_INVALID_INDEX && !list->alloc) {
         return aws_raise_error(AWS_ERROR_LIST_EXCEEDS_MAX_SIZE);
     }
 
+    if (err_code == AWS_OP_SUCCESS) {
+        AWS_POSTCONDITION(aws_array_list_is_valid(list));
+    }
```

Comment:
I believe the array_list remains valid whether or not an error-code is returned from push_back()

likewise for all of these functions

Reply:
Done.